src/main/java/org/openmaptiles/layers/Water.java

```diff
@@ -121,11 +182,122 @@ public void process(Tables.OsmWaterPolygon element, FeatureCollector features) {
         .setAttr(Fields.INTERMITTENT, element.isIntermittent() ? 1 : 0)
         .setAttrWithMinzoom(Fields.BRUNNEL, Utils.brunnel(element.isBridge(), element.isTunnel()), 12)
         .setAttr(Fields.CLASS, clazz);
+
+      try {
+        attemptNeLakeIdMapping(element);
+      } catch (GeometryException e) {
+        e.log(stats, "omt_water",
+          "Unable to add OSM ID to natural earth water feature", config.logJtsExceptions());
+      }
+    }
+  }
+
+  void attemptNeLakeIdMapping(Tables.OsmWaterPolygon element) throws GeometryException {
+    // if OSM lake is too small for Z6 (e.g. area bellow ~4px) we assume there is no matching NE lake
+    var geom = element.source().worldGeometry();
+    if (geom.getArea() < OSM_ID_MATCH_AREA_LIMIT) {
+      return;
+    }
+
+    if (!geom.isValid()) {
+      geom = GeometryFixer.fix(geom);
+      stats.dataError("omt_fix_water_before_ne_intersect");
+      LOGGER.debug("Fixing geometry of OSM element {} before attempt to add ID to natural earth water feature", element.source().id());
```

Comment:
looks like you need to run `./scripts/format.sh`

Reply:
Thanks, adjusted.